### ISSUE TITLE
Consider non-standard regex engine (boost)

### ISSUE THREAD

**Issue by lemire** [ada-url/ada#806]
https://www.boost.org/doc/libs/1_87_0/libs/regex/doc/html/index.html

This is not recommended, but it is an option. Not that boost regex can be used in standalone mode, without all of boost.

**Comment by bbayles** [comment]
Speaking of nonstandard regular expression engines, `re2` would prevent ReDOS exploitation.

**Comment by lemire** [comment]
@bbayles Do you know whether re2 can be conveniently bundled? We release ada as one header file and one source file with no dependencies?

**Comment by lemire** [comment]
I think re2 has non-trivial dependencies. That’s a problem…

**Comment by bbayles** [comment]
It needs [Abseil](https://github.com/abseil/abseil-cpp) as I understand, yes - somewhat akin to Boost, but less common.

**Comment by anonrig** [comment]
@danlapid recommended this https://github.com/hanickadot/compile-time-regular-expressions

**Comment by lemire** [comment]
@anonrig I don't think it supports the full ECMAScript standard, however. If it did (or if it does), then it would be really an alternative to consider.

Note that even with std::regex, there are games we can play.

**Comment by lemire** [comment]
Boost does not support the ECMAScript syntax.
